apps/member-profile/app/routes/integrations.oauth.github.ts

```diff
@@ -0,0 +1,60 @@
+import { type LoaderFunctionArgs, redirect } from '@remix-run/node';
+
+import { ErrorWithContext } from '@/shared/errors';
+import { Route } from '../shared/constants';
+import { ENV } from '../shared/constants.server';
+import {
+  authenticateWithGithub,
+  getGithubProfile,
+  reportError,
+  updateMember,
+} from '../shared/core.server';
+import { ensureUserAuthenticated, user } from '../shared/session.server';
+
+// Note: This serves as a REST endpoint when authenticating with GitHub.
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const session = await ensureUserAuthenticated(request);
+
+  const { searchParams } = new URL(request.url);
+  const { code } = Object.fromEntries(searchParams.entries());
+
+  if (!code || !ENV.GITHUB_OAUTH_CLIENT_ID || !ENV.GITHUB_OAUTH_CLIENT_SECRET) {
+    return redirect(Route['/profile/integrations']);
+  }
+
+  const memberId = user(session);
+
+  try {
+    const { accessToken } = await authenticateWithGithub({
+      clientId: ENV.GITHUB_OAUTH_CLIENT_ID,
+      clientSecret: ENV.GITHUB_OAUTH_CLIENT_SECRET,
+      code,
+    });
+
+    const githubProfile = await getGithubProfile({ accessToken });
```

Comment:
I moved some of the GitHub authentication to the `core` package, specifically the `authenticateWithGithub` and `getGithubProfile` functions. Once we add other integrations, we can consider abstracting this further given that OAuth 2.0 operates the same in most environments.